cookies/giles.py

```diff
@@ -654,10 +687,36 @@ def process_details(data, upload_id, username):
                                      content_type=fmt_data.get('content-type'),
                                      name='%s (%s)' % (page_resource.name, fmt))
 
+        # Content resource for each additional file, if available.
+        for additional_file in page_data.get('additionalFiles', []):
+            content_type = additional_file.get('content-type')
+            uri = '%s/files/%s' % (giles, additional_file.get('id'))
+            resource_type = _get_resource_type(additional_file)
+            content_resource  = _create_content_resource(
+                page_resource,
+                resource_type,
+                creator, uri,
+                _fix_url(additional_file.get('url')),
+                public=False,
+                content_type=content_type,
+                name='%s - %s (%s)' % (page_resource.name,
+                                       additional_file.get('id'),
+                                       content_type)
+            )
+
+            if additional_file.get('processor'):
+                Relation.objects.create(
+                    source=content_resource,
+                    predicate=__creator__,
+                    target=Value.objects.create(name=additional_file.get('processor')),
+                    container=content_resource.container,
+                )
```

Comment:
the logic to get the info for additional pages should be pretty much the same for pages and for the original file. It would make sense to extract a method for that.